### ISSUE TITLE
Test S3 upload gives false negative for new regions

### ISSUE THREAD

**Issue by kingkero** [szepeviktor/fix-w3tc#441]
I am testing this plugin on Ubuntu 16.04, NGINX 1.10.0, PHP 7.0.16-4+deb.sury.org~xenial+1, MariaDB, WP 4.7.3 and Plugin version 0.9.5.

When using (rather) new S3 location - Frankfurt to be precise - the "Test S3 upload" button gives a false negative: "Error: Bucket location region is incorrect. Please select the right one."

False because uploads to that bucket from inside W3TC work without a problem.

**Comment by szepeviktor** [comment]
I don't know but Frankfurt has new signatures.
Could you try with a bucket in Ireland?

**Comment by kingkero** [comment]
Strange thing: I was changing CDN settings (to CloudFront and S3 compatible - then back to S3), now the button gives the proper message of " Test passed ". I don't have time to investigate this right now, will be able to look into this on the weekend.

**Comment by nigrosimone** [comment]
Maybe it's just a random network communication fail, if you try to press the the button it always return success?

**Comment by kingkero** [comment]
Now it does. Might it have been that my first selection of CDN type / S3 settings might not have been saved properly? Or .. my own fault by not flushing db cache after changing settings?

**Comment by nigrosimone** [comment]
when you push the test button, w3tc use the values written in the admin page (also if unsaved). Maybe, your failed test is caused by network error between your website and s3. Sometime it could be happen.

**Comment by lvl99** [comment]
I had the same issue. I originally selected "Amazon CloudFront" and tried to test and it said I had the wrong location selected. When I checked what was sent via the Network tab in Chrome Dev Tools, there was no "config[bucket_location]" value set, so it would error.

To fix I changed my CDN to "S3 (compatible)", saved, then changed it back to "CloudFront", saved, then opened up the CDN options, tested, and it worked with "Test passed".
